bundles/org.eclipse.swt.svg/src/org/eclipse/swt/svg/JSVGRasterizer.java

```diff
@@ -0,0 +1,144 @@
+			KEY_INTERPOLATION, VALUE_INTERPOLATION_BICUBIC, //
+			KEY_RENDERING, VALUE_RENDER_QUALITY, //
+			KEY_STROKE_CONTROL, VALUE_STROKE_PURE, //
+			KEY_TEXT_ANTIALIASING, VALUE_TEXT_ANTIALIAS_ON //
+	);
+
+	@Override
+	public ImageData[] rasterizeSVG(InputStream inputStream, int zoom) throws IOException {
+		SVGDocument svgDocument = loadSVG(inputStream);
+		if (svgDocument != null) {
+			return generateRasterizedImageData(svgDocument, zoom);
+		} else {
+			SWT.error(SWT.ERROR_INVALID_IMAGE);
+		}
+		return null;
+	}
+
+	private SVGDocument loadSVG(InputStream inputStream) {
+		return SVG_LOADER.load(inputStream, null, LoaderContext.createDefault());
+	}
+
+	private ImageData[] generateRasterizedImageData(SVGDocument svgDocument, int zoom) {
+		BufferedImage rasterizedImage = renderSVG(svgDocument, zoom);
+		return convertToSWTImageData(rasterizedImage);
+	}
+
+	private BufferedImage renderSVG(SVGDocument svgDocument, int zoom) {
+		float scalingFactor = zoom / 100.0f;
+		BufferedImage image = createImageBase(svgDocument, scalingFactor);
+		Graphics2D g = configureRenderingOptions(scalingFactor, image);
+		svgDocument.render(null, g);
+		g.dispose();
+		return image;
+	}
+
+	private BufferedImage createImageBase(SVGDocument svgDocument, float scalingFactor) {
+		FloatSize sourceImageSize = svgDocument.size();
+		int targetImageWidth = calculateTargetWidth(scalingFactor, sourceImageSize);
+		int targetImageHeight = calculateTargetHeight(scalingFactor, sourceImageSize);
+		return new BufferedImage(targetImageWidth, targetImageHeight, BufferedImage.TYPE_INT_ARGB);
+	}
+
+	private int calculateTargetWidth(float scalingFactor, FloatSize sourceImageSize) {
+		double sourceImageWidth = sourceImageSize.getWidth();
+		return (int) Math.round(sourceImageWidth * scalingFactor);
+	}
+
+	private int calculateTargetHeight(float scalingFactor, FloatSize sourceImageSize) {
+		double sourceImageHeight = sourceImageSize.getHeight();
+		return (int) Math.round(sourceImageHeight * scalingFactor);
+	}
+
+	private Graphics2D configureRenderingOptions(float scalingFactor, BufferedImage image) {
+		Graphics2D g = image.createGraphics();
+		g.setRenderingHints(RENDERING_HINTS);
+		g.scale(scalingFactor, scalingFactor);
+		return g;
+	}
+	
+	private ImageData[] convertToSWTImageData(BufferedImage rasterizedImage) {
+		int width = rasterizedImage.getWidth();
+	    int height = rasterizedImage.getHeight();
+		int[] pixels = ((DataBufferInt) rasterizedImage.getRaster().getDataBuffer()).getData();
+		PaletteData paletteData = new PaletteData(0x00FF0000, 0x0000FF00, 0x000000FF);
+        ImageData imageData = new ImageData(width, height, 32, paletteData);
+        int index = 0;
+        for (int y = 0; y < imageData.height; y++) {
+			for (int x = 0; x < imageData.width; x++) {
+				int alpha = (pixels[index] >> 24) & 0xFF;
+				imageData.setAlpha(x, y, alpha);
+				imageData.setPixel(x, y, pixels[index++]);
+			}
+		}
+        return new ImageData[]{imageData};
```

Comment:
Some formatting is broken in this class. Please apply auto-formatting.

Reply:
done.